### PR TITLE
Remove env from core package

### PR DESCRIPTION
**Pull request by davidebianchi**

### CODE REVIEW

File: core/opa_transport.go [hunk 1]
```diff
@@ -40,26 +40,40 @@ type OPATransport struct {
 	request                  *http.Request
 	permission               *openapi.RondConfig
 	partialResultsEvaluators PartialResultsEvaluators
-	env                      config.EnvironmentVariables
+
+	clientHeaderKey         string
```
Comment (fredmaggiowski) [inline core/opa_transport.go:44]: can we wrap these options in a Options/RequestOptions struct?

File: internal/config/env_test.go [hunk 2]
```diff
@@ -208,3 +208,19 @@ func TestGetAdditionalHeadersToProxy(t *testing.T) {
...
+func TestIsTraceLogLevel(t *testing.T) {
+	t.Run("true", func(t *testing.T) {
+		env := EnvironmentVariables{
+			LogLevel: traceLogLevel,
+		}
+
+		require.True(t, env.IsTraceLogLevel())
+	})
+
+	t.Run("true", func(t *testing.T) {
```
Comment (fredmaggiowski) [inline internal/config/env_test.go:221]:
```suggestion
	t.Run("false", func(t *testing.T) {
```

File: openapi/openapi_utils.go [hunk 3]
```diff
@@ -330,7 +335,7 @@ func LoadOASFromFileOrNetwork(log *logrus.Logger, env config.EnvironmentVariable
 		return oas, nil
 	}
 
-	return nil, fmt.Errorf("missing environment variables one of %s or %s is required", config.TargetServiceOASPathEnvKey, config.APIPermissionsFilePathEnvKey)
+	return nil, fmt.Errorf("missing openapi config: one of TargetServiceOASPath or APIPermissionsFilePath is required")
```
Comment (fredmaggiowski) [inline openapi/openapi_utils.go:338]: The problem with this change in the error is that it will be logged with an incomplete information for the user; it will see the name of the field instead of the env var that should be set.

The error in itself is fine, but I'd wrap it and make a check in the main function to provide further information to the user about the env var that are missing.

File: types/rbactypes.go [hunk 4]
```diff
@@ -23,6 +23,7 @@ type User struct {
 	UserGroups   []string
 	UserRoles    []Role
 	UserBindings []Binding
+	Properties   map[string]any
```
Comment (fredmaggiowski) [inline types/rbactypes.go:26]: Are we adding this new feature in the PR? This is an important feature; please change the PR title and description to reflect this change so that it gets its proper space in the release (or move it to another PR altogether)

File: core/opaevaluator.go [hunk 5]
```diff
@@ -365,65 +378,19 @@ func (evaluator *OPAEvaluator) PolicyEvaluation(logger *logrus.Entry, permission
...
-			Bindings:               user.UserBindings,
-			Roles:                  user.UserRoles,
-			Properties:             userProperties,
-			Groups:                 userGroup,
-			ResourcePermissionsMap: permissionsMap,
-		},
-	}
+func CreateRegoQueryInput(
+	logger *logrus.Entry,
+	input Input,
```
Comment (fredmaggiowski) [inline core/opaevaluator.go:387]: I'm not convinced about separating input creation from the rego input creation; they are the same and I'd require the SDK user to invoke two functions rather than one. 

Your idea was to make InputFromRequest interchangeable with other InputFromFiberRequest InputFromSomethingElse, etc?

If that's the case I would not create different functions but create a single function that accepts an interface and if you want to build the input from something else you have to implement the interface for that (rather than duplicating the input creation). This way I'd stick with a single function `CreateRegoQueryInput` (which i'd rename to `CreateRegoInput` or simply `CreateInput`) which accepts an `InputSource` interface (we can provide some ready to use implementations, e.g for `*http.Request`)

**Review by fredmaggiowski — CHANGES_REQUESTED** [review]

File: types/rbactypes.go (hunk 4 above)
Comment (davidebianchi) [reply types/rbactypes.go:26]: It's not a new feature, it's a cleanup. Before, properties are handled only in CreateRegoInput https://github.com/rond-authz/rond/blob/829a3dd25588aaf91038e7cea1ab9de4e41d8830/core/opaevaluator.go#L372-L376.

But this information is related to the user, as userId and groups which are already taken in the `RetrieveUserBindingsAndRoles` function.

File: core/opaevaluator.go (hunk 5 above)
Comment (davidebianchi) [reply core/opaevaluator.go:387]: Yes, my idea is to make InputFromRequest interchangeable.
At the moment, this is the cleaner way to do it as first step to avoid a huge PR. The basic idea is to create the sdk and I'm not sure at the moment if the core package should manage the input creation or it should be done by sdk.

At the moment the scope is to separate the functions to better understand the different ownership.

File: core/opa_transport.go (hunk 1 above)
Comment (davidebianchi) [reply core/opa_transport.go]: Done

File: types/rbactypes.go (hunk 4 above)
Comment (fredmaggiowski) [reply types/rbactypes.go:26]: They weren't put in the rego input, though... Am I missing something? 🤔

File: core/opaevaluator.go (hunk 5 above)
Comment (fredmaggiowski) [reply core/opaevaluator.go:387]: mmmh not yet convinced but ok

File: openapi/openapi_utils.go (hunk 3 above)
Comment (davidebianchi) [reply openapi/openapi_utils.go:338]: Added in the config package

File: types/rbactypes.go (hunk 4 above)
Comment (davidebianchi) [reply types/rbactypes.go:26]: This is the entire actual function https://github.com/rond-authz/rond/blob/829a3dd25588aaf91038e7cea1ab9de4e41d8830/core/opaevaluator.go#L368-L433

At line 407, permissions are set in the input user

**Review by fredmaggiowski — APPROVED** [review]

**Review by fredmaggiowski — APPROVED** [review]
